File: custom_data_structures/js_data_structures/node.js
var Node = function(value){
  // this section is used to give each node its own unique id
  // the delay (while loop) is necessary because otherwise many
  // nodes could be created w/in the same milisecond and break the
  // unique constraint.
  // TODO: find a less hacky way to gurantee a unique ID
  // ********************************************************
  var timer = Date.now();
  while (Date.now() === timer){}
  this.id = Date.now();
  // ********************************************************
  this.value = value;
  this.next = null;
};

File: custom_data_structures/ruby_data_structures/node.rb
class Node
  attr_accessor :value, :next

  def initialize value=nil
    @value = value
    @next = nil
  end

  def to_s
    p "Node  value: #{self.value ? value : 'no value set'}  " +
    "next: #{self.next ? self.value : 'no next node'}"
  end
end